scikeras/wrappers.py

```diff
@@ -697,26 +697,41 @@ def feature_encoder(self):
         """
         return FunctionTransformer()
 
-    def fit(self, X, y, sample_weight=None) -> "BaseWrapper":
+    def fit(self, X, y, sample_weight=None, **kwargs) -> "BaseWrapper":
         """Constructs a new model with ``model`` & fit the model to ``(X, y)``.
 
         Parameters
         ----------
-        X : Union[array-like, sparse matrix, dataframe]of shape (n_samples, n_features)
+        X : Union[array-like, sparse matrix, dataframe] of shape (n_samples, n_features)
             Training samples, where n_samples is the number of samples
             and n_features is the number of features.
         y : Union[array-like, sparse matrix, dataframe] of shape (n_samples,) or (n_samples, n_outputs)
             True labels for X.
         sample_weight : array-like of shape (n_samples,), default=None
             Array of weights that are assigned to individual samples.
             If not provided, then each sample is given unit weight.
-
+        **kwargs : Dict[str, Any]
+            Extra arguments to route to ``Model.fit``.
+            This paremeter will be removed in a future release of SciKeras,
+            instead set fit arguments at initialization
+            (e.g., ``BaseWrapper(epochs=10, ...)``)
         Returns
         -------
         BaseWrapper
             A reference to the instance that can be chain called
             (ex: instance.fit(X,y).transform(X) )
         """
+        for k, v in kwargs.items():
+            warnings.warn(
+                "``kwargs`` will be removed in a future release of SciKeras."
+                f"Instead, set fit arguments at initialization (i.e., ``BaseWrapper({k}={v})``)"
+            )
+        kwargs = {
+            (k if k.startswith("fit__") else "fit__" + k): v for k, v in kwargs.items()
+        }
+        existing_kwargs = {k: v for k, v in self.get_params().items() if k in kwargs}
+        self.set_params(**kwargs)
```

Comment:
:-1:

I'm still not seeing the need to set parameters. It looks like `set_params` is a vehicle to get these `kwargs` to `self._fit` or `self._fit_keras_model`. Why can't `kwargs` be passed to those functions and this processing be done there?

Reply:
You were the one who originally suggested this approach in https://github.com/adriangb/scikeras/issues/124#issuecomment-723143535

We could pass `**kwargs` around. Perhaps at this level of complexity/LOC that is a better approach. I'll try implementing it that way instead and see how it turns out.

Reply:
I was able to implement this passing `**kwargs` around directly. It is indeed simpler and cleaner.